api/uniter/unit.go

```diff
@@ -304,29 +304,30 @@ func (u *Unit) AssignedMachine() (names.MachineTag, error) {
 	return names.ParseMachineTag(result.Result)
 }
 
-// IsPrincipal returns whether the unit is deployed in its own container,
-// and can therefore have subordinate services deployed alongside it.
+// PrincipalUnit returns whether the the principal unit name as well
+// as whether the current unit is principal, and can therefore have
+// subordinate services deployed alongside it.
 //
 // NOTE: This differs from state.Unit.IsPrincipal() by returning an
 // error as well, because it needs to make an API call.
-func (u *Unit) IsPrincipal() (bool, error) {
+func (u *Unit) PrincipalUnit() (string, bool, error) {
 	var results params.StringBoolResults
 	args := params.Entities{
 		Entities: []params.Entity{{Tag: u.tag.String()}},
 	}
 	err := u.st.facade.FacadeCall("GetPrincipal", args, &results)
 	if err != nil {
-		return false, err
+		return "", false, err
 	}
 	if len(results.Results) != 1 {
-		return false, fmt.Errorf("expected 1 result, got %d", len(results.Results))
+		return "", false, fmt.Errorf("expected 1 result, got %d", len(results.Results))
 	}
 	result := results.Results[0]
 	if result.Error != nil {
-		return false, result.Error
+		return "", false, result.Error
 	}
 	// GetPrincipal returns false when the unit is subordinate.
```

Comment:
Just thinking a bit more, I'd prefer if PrincipalName mirrored the backend code, so it returns true for a subordinate (principal name != ""), and false for principal (principal name == ""). The existing caller of IsPrincipal/PrincipalName would need to invert its check.

If we leave the apiserver facade alone, here you would do something like:

```go
var unitName string
if result.Ok {
    unitTag, err := names.ParseUnitTag(result.Result)
    if err != nil {
        return "", false, err
    }
    unitName = unitTag.Id()
}
return unitName, result.Ok, nil
```